File: ruby/ShieldBooster.rb
# Representa a los potenciadores de escudo que pueden tener las estaciones
# espaciales

require_relative 'ShieldToUI'

module Deepspace
    class ShieldBooster
        def initialize(initName, initBoost, initUses)
            @name = initName
            @boost = initBoost
            @uses = initUses
        end

        # Constructor de copia
        def self.newCopy(other)
            new(other.name, other.boost, other.uses)
        end

        def name
            @name
        end

        def boost
            @boost
        end

        def uses
            @uses
        end

        def useIt
            if @uses > 0
                @uses-=1
                @boost
            else
                1.0
            end
        end

        def getUIversion
            ShieldToUI.new(self)
        end

        def to_s
            "name: #@name, boost: #@boost, uses: #@uses";
        end
    end
end

File: java/DeepSpace/src/deepspace/ShieldBooster.java
package deepspace;


// Representa a los potenciadores de escudo que pueden tener las estaciones
// espaciales
class ShieldBooster implements CombatElement {
    private String name;
    private float boost;
    private int uses;
    
    ShieldBooster(String name, float boost, int uses){
        this.name = name;
        this.boost = boost;
        this.uses = uses;
    }
    
    ShieldBooster(ShieldBooster other){
        this.name = other.name;
        this.boost = other.boost;
        this.uses = other.uses;
    }
    
    public float getBoost(){
        return boost;
    }
    
    @Override
    public int getUses(){
        return uses;
    }
    
    @Override
    public float useIt(){
        if (uses > 0){
            uses--;
            return boost;
        }
        else
            return 1.0f;
    }
    
    ShieldToUI getUIversion(){
        return new ShieldToUI(this);
    }
    
    public String toString(){
        return "name: " + name +
                ", boost: " + boost +
                ", uses: " + uses;
    }
}
